src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -101,7 +101,9 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 
             var tagHelpersFromReferences = context.CompilationProvider
                 .Combine(razorSourceGeneratorOptions)
-                .WithLambdaComparer(static (a, b) => a.Left.References.SequenceEqual(b.Left.References), static a => a.Left.References.GetHashCode())
+                .WithLambdaComparer(
```

Comment:
So the issue was that the options were being messed up in the past? And now this code makes sure that the options are properly merged?

Reply:
Each `Select` is a step in in the incremental source generator. In this step, our inputs are the `Compilation + RazorSourceGeneratorOptions` and we produce a list of `TagHelperDescriptor` that are discovered from the reference assemblies. 

On Line 111, we skip doing any work if the source generator is suppressed. This happens to avoid runtime code messing up razor tooling during design time builds. The baseline compilation in VS runs with `SuppressRazorSourceGenerator=true`. In a hot reload build, the compiler flips `SuppressRazorSourceGenerator=false` and re-invokes the source generator. The source generator infrastructure in Roslyn would invoke the comparer and figure out if it needs to re-compute. Unfortunately, the previous code only varied by the list of reference assemblies, which did not change between these compilations. So it re-uses the previously cached empty results and the hot reload compilation fails to process any tag helpers from ref assemblies.